agent/metadata_service.go

```diff
@@ -34,6 +34,7 @@ type MetadataService interface {
 	Port() int
 }
 
+// CredentialsSource all we need to get STS credentials
```

Comment:
This comment doesn't really add value.

Reply:
Any `Upercased` symbol needs that sort of docstring. Sadly I had nothing better to say.